core/src/main/java/org/apache/iceberg/MetadataUpdateParser.java

```diff
@@ -72,6 +78,32 @@ private MetadataUpdateParser() {
   // AddSortOrder
   private static final String SORT_ORDER = "sort-order";
 
+  // SetDefaultSortOrder
+  private static final String SORT_ORDER_ID = "sort-order-id";
+
+  // AddSnapshot
+  private static final String SNAPSHOT = "snapshot";
+
+  // RemoveSnapshot
+  private static final String SNAPSHOT_IDS = "snapshot-ids";
+
+  // SetSnapshotRef
+  private static final String REF_NAME = "ref";
```

Comment:
I'd support updating this to `ref-name` here and in the spec.

Reply:
+1. Do we also want to change the corresponding update requirement field? We had the same concern there.

https://github.com/apache/iceberg/blob/3586e140dc3fbe0b0414464497be3ccd4ca0927d/core/src/main/java/org/apache/iceberg/rest/requests/UpdateRequirementParser.java#L55

Reply:
Here's the associated class `UpdateRequirement.AsserrRefSnapshotId`: https://github.com/apache/iceberg/blob/3586e140dc3fbe0b0414464497be3ccd4ca0927d/core/src/main/java/org/apache/iceberg/rest/requests/UpdateTableRequest.java#L261-L272

Reply:
I updated this to `ref-name`, as well as updated the spec.

I also updated the code to use `refName` as the field in `SetSnapshotRef` as well as the accessor (to match `AssertRefSnapshotId`. Please let me know if you think the changes for the accessor are too much for this PR.